### PR TITLE
handle token type

### PR DESCRIPTION
**Pull request by mauerbac**
how does this look @jgaul

### CODE REVIEW

File: optimizely/client.py [hunk 1]
```diff
@@ -12,10 +12,11 @@
 class Client(object):
   ALLOWED_REQUESTS = ['get', 'post', 'put', 'delete']
 
-  def __init__(self, api_key, api_base=BASE_URL):
+  def __init__(self, api_key, api_base=BASE_URL, token_type):
```
Comment (jgaul) [inline optimizely/client.py]: positional args should be before any keyword args. i don't think this code will even run, will it?

File: optimizely/client.py [hunk 2]
```diff
@@ -12,10 +12,11 @@
...
     # set API information
     self.api_key = api_key
     self.api_base = api_base
+    self.token_type = token_type
```
Comment (jgaul) [inline optimizely/client.py]: also, there should be validation that the provided type is one of the 2 supported types

Comment (jgaul) [inline optimizely/client.py]: i see you have that below, but it's better to error as soon as we know something's wrong.

File: optimizely/client.py [hunk 3]
```diff
@@ -34,10 +35,12 @@ def request(self, method, url_parts, headers=None, data=None):
...
 
       # test if Oauth token
-      if ":" in self.api_key:
+      if self.token_type == 'legacy':
         headers.update({'Token': self.api_key, 'User-Agent': 'optimizely-client-python/0.1.1'})
-      else:
+      elif self.token_type == 'oauth':
         headers.update({'Authorization': ' Bearer ' + self.api_key, 'User-Agent': 'optimizely-client-python/0.1.1'})
+      else:
+        raise error.BadRequestError('%s is not a valid token type.' % token_type)
```
Comment (jgaul) [inline optimizely/client.py]: `BadRequestError` wouldn't be appropriate here since it's not an HTTP error

**Comment by mauerbac** [comment]
Addressed all three concerns. How is this? @jgaul

File: optimizely/client.py [hunk 4]
```diff
@@ -8,15 +8,21 @@
...
-  def __init__(self, api_key, api_base=BASE_URL):
+  def __init__(self, api_key, token_type, api_base=BASE_URL):
     # set API information
     self.api_key = api_key
     self.api_base = api_base
 
+    if token_type in VALID_TOKEN_TYPES:
+      self.token_type = token_type
+    else:
+      raise ValueError('Invalid Token Type')
```
Comment (jgaul) [inline optimizely/client.py]: would be good to enumerate the valid token types in this error. just something like raise ValueError('Invalid token type. Valid types are: %s' % (VALID_TOKEN_TYPES))

**Pull request merged**